### PR TITLE
Write unix timestamp of last play to music file metadata

### LINKED ISSUE #21: Ensure, that songs are not played more than once a day

### PR DESCRIPTION
**Pull request by smlz**
This should simplify checking when the song was last played in the player, and might help with issue #21.

### CODE REVIEW

File: klangbecken.py [hunk 1]
```diff
@@ -1041,9 +1043,16 @@ def playlog_cmd(data_dir, filename, off_air=False, dev_mode=False):
         json.dump(data, f, **json_opts)
         del data
 
+    # Update file metadata
+    FileType = SUPPORTED_FILE_TYPES[filename.split(".")[-1]]
```
Comment (spameier) [inline klangbecken.py:1047]: `SUPPORTED_FILE_TYPES` is defined with leading dots so you need to include that here as well (`FileType = SUPPORTED_FILE_TYPES["." + filename.split(".")[-1]]`).

File: klangbecken.py [hunk 2]
```diff
@@ -1041,9 +1043,16 @@ def playlog_cmd(data_dir, filename, off_air=False, dev_mode=False):
...
+    mutagenfile = FileType(filename)
+    mutagenfile["last_play"] = str(now.timestamp())
+
```
Comment (spameier) [inline klangbecken.py:1050]: Save the file after modifying (`mutagenfile.save()`).

File: klangbecken.py (hunk 1 above)
Comment (smlz) [reply klangbecken.py]: yep, it is a while ago... thanks

File: klangbecken.py (hunk 2 above)
Comment (smlz) [reply klangbecken.py:1047]: Thanks, fixed it.

**Comment by smlz** [comment]
Just a remark as we were wondering yesterday, how it comes that we have so many songs played twice a day.

At first it seemed to me very improbable that this can happen, with 1500 songs in the playlist. But as it turns out, it is not. During an afternoon with about one hundred song plays, with every song picked completely at random, the probability to have at least one song repetition is a little over 96%. So when during an afternoon the playlist is edited a lot, by uploading new songs, and adjusting the playlist, the playlist gets shuffled and reloaded with every such edit. This makes picking the next song often completely random, instead of going through a randomized playlist. So seeing repetitions is actually _no_ surprise.

With a bit of math, this behavior could actually have been predicted, and is a nice example of the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem).

**Comment by smlz** [comment]
Surprisingly, with just 46 song plays, the probability of having a repetition goes over 50 percent. So much for the :1234: nerd stats...

**Comment by spameier** [comment]
Looks good! The recorded timestamp can now be extracted in `liquidsoap` like so: 
```
....
metadata = request.metadata(r)
last_play = metadata["last_play"]
...
```

**Pull request merged**